### ISSUE TITLE
scrolling in text editors like nano/vim is jittery/jerky, framerate seems laggy

### ISSUE THREAD

**Issue by insidewhy** [zellij-org/zellij#3793]
`zellij --version`: zellij 0.41.1

`stty size`: 34 116

`uname -av`: Linux rats 6.11.8-arch1-2 #1 SMP PREEMPT_DYNAMIC Fri, 15 Nov 2024 15:35:07 +0000 x86_64 GNU/Linux

## Issue description

Scrolling in neovim by holding `j` or `k` looks choppy, or in `nano` by holding `<up>` or `<down>`, like the framerate is tanking. Outside of zellij everything looks smooth and consistent. This happens with an empty (default) neovim configuration (e.g. `nvim --clean`) but given it happens in `nano` also I guess it's not specific to any editor or app with fast scrolling.

## Minimal reproduction

1. Run `nvim --clean <large file>`
2. Scroll downwards by holding `j`.
3. Notice the scroll is choppy and much slower than scrolling outside of neovim.
4. Notice the same thing happens when using `nano <large file>` and holding the down arrow.

## Other relevant information

Rendering within neovim in general seems quite a lot slower, for example with `:set showcmd`, outside of `zellij` the updates are so quick you never even glimpse the echod commands, but with `zellij` they are commonly visible.

I used `time nvim <large file>` and scrolled to the bottom before exiting, and the times were approximately the same whether inside of `zellij` or not, so it only seems the rendering is affected and not the actual speed of scrolling.

This issue does not occur with `tmux`.

**Comment by insidewhy** [comment]
The render performance of zellij was a little below what I could tolerate, so I spent a while customising kitty instead and will use tmux when I have to for now, it's inconvenient that it doesn't yet support the kitty keyboard protocol but at least the render performance is adequate for me.

I wrote a kitten to forward my next/previous/create tab arguments to tmux if it's running, otherwise use native kitty tabs. And some other python to customise the kitty tab bar. So now whether I'm using kitty, tmux or zellij everything looks the same.

**Comment by koalazub** [comment]
Is this related to #2548? Would merging this help address issues related to scrolling

**Comment by insidewhy** [comment]
> Is this related to #2548? Would merging this help address issues related to scrolling

I have no idea, I'll try it

**Comment by insidewhy** [comment]
@koalazub Yeah that really improve performance for me. Oh no it's been left unmerged since more than a year ago :(

**Comment by valankar** [comment]
Will this issue ever be fixed? Tried Zellij again after about a year and it is still unbearably slow at scrolling compared to any other multiplexer (tmux, screen). I really like it but just can't use it.

**Comment by insidewhy** [comment]
@valankar Agree with you on that, you can get more context in https://github.com/zellij-org/zellij/pull/2548 although if you read this comment https://github.com/zellij-org/zellij/pull/2548#issuecomment-2487889708 the author explicitly stated they won't engage with the issue anymore, so it seems I may have accidentally sabotaged the chances of this getting fixed by unintentionally causing some drama by @-ing the maintainer on an issue I felt was very important and neglected.
